dev/plugins/hu.elte.txtuml.xtxtuml.ui/src/hu/elte/txtuml/xtxtuml/ui/contentassist/XtxtUMLReferenceProposalCreator.xtend

```diff
@@ -274,16 +271,12 @@ class XtxtUMLReferenceProposalCreator extends XbaseReferenceProposalCreator {
 			// be difficult to achieve starting from a plain JvmType
 			proposedObj instanceof JvmDeclaredType && (proposedObj as JvmDeclaredType).superTypes.exists [
 				val typeRef = toLightweightTypeReference;
-				typeRef.isSubtypeOf(DataType) || typeRef.isInterface && typeRef.isSubtypeOf(ExternalType) || typeRef.isSubtypeOf(ModelEnum) ||
```

Comment:
I meant your first interpretation. I think it's enough if we check the _model_ (the part of the code where we are about to create proposals) using the externality helper, but it will require some testing as I'm not sure if the model is just a proxy here, see #507. I believe this modification could be placed right before _[XtxtUMLReferenceProposalCreator.xtend:245](https://github.com/ELTE-Soft/txtUML/blob/1d5ff11b2d8a354b1a90e861ced5ef50f9c6aef9/dev/plugins/hu.elte.txtuml.xtxtuml.ui/src/hu/elte/txtuml/xtxtuml/ui/contentassist/XtxtUMLReferenceProposalCreator.xtend#L245)_.